### ISSUE TITLE
The modified value cannot be hexadecimal

### ISSUE THREAD

**Issue by shing-lee** [mikefarah/yq#929]
**Describe the bug**
Given a file called `config.yaml` containing the following YAML:
```
version: 1.0.1
id: 0x1111
```

Processing with this shell:
```
1. yq e '.id=0x2222' config.yaml
2. yq e '.id=“0x2222”' config.yaml
```

Output is unfortunately:
```
1:
Error: Parsing expression: Lexer error: could not match text starting at 1:6 failing at 1:7.
	unmatched text: "x"
```
```
2:
version: 1.0.1
id: ”0x2222“
```

Expected output:

```
version: 1.0.1
id: 0x2222
```

Version of yq: 4.12.1
Operating system: linux
Installed via: binary release

**Comment by mikefarah** [comment]
Will fix in next release

**Comment by mikefarah** [comment]
Fixed in 4.12.2
